website/mails/mails.py

```diff
@@ -117,6 +123,53 @@ def send_mail(to_addr, mail, mimetype='plain', from_addr=None, mailer=None,
 
         return ret
 
+def queue_mail(to_addr, mail, send_at, user, **context):
+    new_mail = QueuedMail()
+    new_mail.create(to_addr, mail, send_at, user, **context)
+
+class QueuedMail(StoredObject):
+    _id = fields.StringField(primary=True, default=lambda: str(bson.ObjectId()))
+    user = fields.ForeignField('User')
+    to_addr = fields.StringField(default=None)
+    send_at = fields.DateTimeField(default=None)
+    email_type = fields.StringField(default=None)
+    data = fields.DictionaryField(default=None)
+    sent_at = fields.DateTimeField(index=True, default=None)
+
+    def create(self, to_addr, mail, send_at, user, **context):
+        self.user = user
+        self.to_addr = to_addr
+        self.send_at = send_at
+        self.email_type = mail['template']
+        self.data = context
+        self.save()
+
+    def send_mail(self):
+        mail_struct = queue_mail_types[self.email_type]
+        callback = mail_struct['callback'](self)
+        mail = Mail(
+            mail_struct['template'],
+            subject=mail_struct['subject']
+        )
+        if callback and self.user.osf_mailing_lists.get('Open Science Framework Help'):
+            send_mail(self.to_addr, mail, mimetype='html', **(self.data or {}))
+            self.sent_at = datetime.utcnow()
+            self.save()
+            return True
+        else:
+            self.delete()
+            return False
+
+    def find_same_sent(self):
+        return list(self.__class__.find(
+            Q('email_type', 'eq', self.email_type) &
+            Q('user', 'eq', self.user) &
+            Q('sent_at', 'ne', None)
+        ))
+
+    def delete(self):
```

Comment:
This is used on one of the callbacks. I added the method to prevent another import, it also seemed universal and useful enough that it could be used by future callbacks.